src/java.base/share/classes/java/math/MutableBigInteger.java

```diff
@@ -1892,6 +1913,183 @@ private boolean unsignedLongCompare(long one, long two) {
         return (one+Long.MIN_VALUE) > (two+Long.MIN_VALUE);
     }
 
+    /**
+     * Calculate the integer {@code n}th root {@code floor(nthRoot(this, n))} and the remainder,
+     * where {@code nthRoot(., n)} denotes the mathematical {@code n}th root.
+     * The contents of {@code this} are <em>not</em> changed. The value of {@code this}
+     * is assumed to be non-negative and the root degree {@code n >= 3}.
+     *
+     * @implNote The implementation is based on the material in Richard P. Brent
+     * and Paul Zimmermann, <a href="https://maths-people.anu.edu.au/~brent/pd/mca-cup-0.5.9.pdf">
+     * Modern Computer Arithmetic</a>, p. 27-28.
+     *
+     * @param n the root degree
+     * @return the integer {@code n}th root of {@code this} and the remainder
+     */
+    MutableBigInteger[] nthRootRem(final int n) {
```

Comment:
We rarely use `final` on a parameter.

Reply:
I guess most developer use a Java IDE for Java development.
My IDE shows variables that are altered by underlying their name at every occurrence in the code. I can see at a glance that `n` is not modified even without `final`.
As said, it's a matter of convention. In the OpenJDK we usually don't use `final` on parameters, as it clutters the header. Just have a look at a random place in the codebase.

Reply:
underlying -> underlining